app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardController.kt

```diff
@@ -408,25 +411,29 @@ class DashboardController(
                 .height(16)
                 .addTo(modelList)
 
-            ButtonType2Model_(onCheckSymptomsAgainClick)
+            ButtonType2Model_ { onCheckSymptomsAgainClick(false) }
                 .id("own_health_status_check_symptoms_button")
                 .text(context.string(R.string.self_testing_symptoms_secondary_button))
                 .enabled(exposureNotificationPhase.isReportingEnabled())
-                .onDisabledClick { onFeelingClick(true) }
+                .onDisabledClick { onCheckSymptomsAgainClick(true) }
                 .addTo(modelList)
         }
 
-        if (ownHealthStatus is HealthStatusData.SicknessCertificate) {
+        val isRedRevokingEnabled = dateOfFirstMedicalConfirmation
```

Comment:
We need to let the user revoke for two full days except today. If he submits the report on 13th June, 15:30, he will not be able to revoke starting from 16th June, 00:00 (beginning of the day).

Reply:
Updated.